server/src/test/java/org/apache/druid/server/coordination/SegmentLoadDropHandlerTest.java

```diff
@@ -567,6 +568,48 @@ public void testProcessBatchDuplicateLoadRequestsWhenFirstRequestFailsSecondRequ
     segmentLoadDropHandler.stop();
   }
 
+  @Test(timeout = 60_000L)
+  public void testProcessBatchWithDefaultNoStorageLocations() throws Exception
+  {
+    SegmentLoadDropHandler handler = new SegmentLoadDropHandler(
+        jsonMapper,
+        new SegmentLoaderConfig(),
+        announcer,
+        Mockito.mock(DataSegmentServerAnnouncer.class),
+        segmentManager,
+        segmentCacheManager,
+        scheduledExecutorFactory.create(5, "SegmentLoadDropHandlerTest-[%d]"),
+        new ServerTypeConfig(ServerType.HISTORICAL)
+    );
+
+    handler.start();
+
+    final DataSegment segment = makeSegment("test", "1", Intervals.of("P1d/2011-04-01"));
+
+    List<DataSegmentChangeRequest> batch = ImmutableList.of(new SegmentChangeRequestLoad(segment));
+
+    ListenableFuture<List<DataSegmentChangeResponse>> future = handler.processBatch(batch);
+
+    for (Runnable runnable : scheduledRunnable) {
+      runnable.run();
+    }
+
+    List<DataSegmentChangeResponse> result = future.get();
+    Assert.assertEquals(1, result.size());
+
+    SegmentChangeStatus expectedFailedStatus = SegmentChangeStatus.failed(
```

Comment:
It doesn't make sense to fail every segment load request. The historical should just fail to start up in the first place.

One way to do that would be to add a constructor to `SegmentLoaderConfig` and do the validation there. Another method could be to add a `validateLocations()` method to `SegmentLoaderConfig` and invoke it from a (probably guice-injected) class which is initialized on startup.